### ISSUE TITLE
allow opening local URIs without opening the browser

### ISSUE THREAD

**Issue by matifali** [coder/coder#9752]
Allow the `coder_app` resource to open local URIs like

- jetbrains-gateway://
- vscode://
- vscode-insiders://

Directly on the client machine without opening the browser. 

This is needed for a better UX when we add external apps using terraform modules.
Related: https://github.com/coder/testing-modules/pull/14 and #9679

I purpose we have something like

```hcl
resource "coder_app" "local_app" {
	url   = jetbrains-gateway://....
	local = true
}
```
or
This is a breaking change
```hcl
resource "coder_app" "local_app" {
	url  = jetbrains-gateway://....
	type = "local" # other types "external", "internal/coder"
}
```

**Comment by bpmct** [comment]
the benefit of this is that currently local apps open in a new window in the browser

**Comment by fcsonline** [comment]
We have this pull request open to achieve this behaviour: https://github.com/coder/coder/pull/9604

**Comment by matifali** [comment]
This is fixed in #9878
